src/bin/install.sh

```diff
@@ -0,0 +1,175 @@
+    fi
+fi
+
+# Determine the installation directory
+INSTALL_DIR="$HOME/.local/bin"
+
+# Check if the installation directory exists and is writable
+if [ ! -d "$INSTALL_DIR" ]; then
+    echo "The installation directory ($INSTALL_DIR) does not exist. Creating it now..."
+    if ! mkdir -p "$INSTALL_DIR"; then
+        echo "Failed to create the installation directory. Please check your permissions and try again."
+        exit 1
+    fi
+elif [ ! -w "$INSTALL_DIR" ]; then
+    echo "The installation directory ($INSTALL_DIR) is not writable. Please check your permissions and try again."
+    exit 1
+fi
+
+# Assuming the binary or application name is predictable and consistent
+BINARY_NAME='defang' # Adjust this based on actual content
+
+# Move the binary or application to the installation directory from the temporary directory
+echo "Moving defang to $INSTALL_DIR"
+if ! mv "$EXTRACT_DIR/$BINARY_NAME" "$INSTALL_DIR"; then
+    echo "Failed to move defang. Please check your permissions and try again."
+    exit 1
+fi
+
+# Make the binary executable
+if ! chmod +x "$INSTALL_DIR/$BINARY_NAME"; then
+    echo "Failed to make defang executable. Please check your permissions and try again."
+    exit 1
+fi
+
+# Cleanup: Remove the temporary directory
+echo "Cleaning up..."
+rm -rf "$EXTRACT_DIR"
+
+# Add the installation directory to PATH if not already present
+if [[ ":$PATH:" != *":$INSTALL_DIR:"* ]]; then
+    echo "Adding $INSTALL_DIR to your PATH for this session."
+    export PATH="$PATH:$INSTALL_DIR"
+
+    # Define the possible shell profile files
+    PROFILE_FILES=(".bashrc" ".zshrc" ".kshrc")
+    FOUND_PROFILE_FILE=false
+
+    # Loop over the possible profile files
+    for profile_file in "${PROFILE_FILES[@]}"; do
+        # If the profile file exists in the user's home directory
+        if [[ -f "$HOME/$profile_file" ]]; then
+            FOUND_PROFILE_FILE=true
+            # Prompt the user for confirmation
+            echo -n "Can we append the necessary line to $HOME/$profile_file? (y/n) "
+            read REPLY 
+            echo    # move to a new line
+            if [[ $REPLY =~ ^[Yy]$ ]]; then
+                # Append the line to the profile file
+                echo "export PATH=\"\$PATH:$INSTALL_DIR\" # Added by Defang Installer" >> "$HOME/$profile_file"
+            fi
+        fi
+    done
+
+    # If no profile file was found
+    if [[ $FOUND_PROFILE_FILE == false ]]; then
+        # Get the name of the current shell
+        CURRENT_SHELL=$(basename "$SHELL")
+        # Prompt the user to create a new profile file
+        echo -n "No existing profile file found. Would you like to create a .$CURRENT_SHELL"rc" file? (y/n) "
+        read REPLY
+        echo    # move to a new line
+        if [[ $REPLY =~ ^[Yy]$ ]]; then
+            # Create the new profile file and append the line
+            echo "export PATH=\"\$PATH:$INSTALL_DIR\"" >> "$HOME/.$CURRENT_SHELL"rc" # Added by Defang Installer"
+        fi
+    fi
```

Comment:
Great check for if the installed file is in $PATH, but do you think it is a bit invasive to modify the customer's rc and profile files or is this a common practice now? I think a good middle ground would be detect it and display the warning and the instructions/one liner command to add it to path.

Reply:
@edwardrf I think that's fair. It does feel a bit invasive. I was a bit hesitant about it, but I also thought it might have been helpful for Hongchen the other day. I figure the prompt gives them a way to opt out. Could always print the instruction if they type `n`?